### PR TITLE
Server Config Parser class additions

### PR DESCRIPTION
**Pull request by alfredlucero**
Added NginxServerConfigParser class that parses a config file for server settings (i.e. port) in a server_config struct and request handlers (static file serve and echo) in a std::vector<std::unique_ptr<http::handler> > (Thank you Brandon for fixing the compiler errors) with some unit tests for port number and parsing handlers

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,7 +25,7 @@ TESTFLAGS=-std=c++11 -isystem ${GTEST_DIR}/include -pthread
 # Source files
 SRC=main.cc server.cc config_parser.cc http_response.cc \
 server_config_parser.cc http_request_parser.cc http_handler_echo.cc \
-http_handler_file.cc
+http_handler_file.cc
```
Comment (gotitmemorize) [inline Makefile]: extra space somehow got added again, so remove it

File: main.cc [hunk 2]
```diff
@@ -17,30 +17,26 @@ int main(int argc, char* argv[]) {
...
         }
+	
+        // Parse the config file into statements and tokens
+        NginxServerConfigParser server_parser(argv[1]);
 
-        // Get the port number from the config file
-        int port = port_number(argv[1]);
+        // Get the server settings from the config file (i.e. port)
+        server_config server_settings;
+        int port = server_parser.ParseServerSettings(server_settings);
```
Comment (gotitmemorize) [inline main.cc]: Just realized that the port that is passed into the server is a short, so maybe change this from an int to a short

File: server_config_parser.h [hunk 3]
```diff
@@ -2,11 +2,37 @@
 #define SERVER_CONFIG_PARSER_H
 
 #include "config_parser.h"
+#include "http_handler_echo.h"
+#include "http_handler_file.h"
 
+struct server_config {
+    int port;
```
Comment (gotitmemorize) [inline server_config_parser.h]: if you change the other instance of port for the server into a short, change this into a short as well

**Comment by gotitmemorize** [comment]
RIP, time to change all the shorts to ints

File: server_config_parser_test.cc [hunk 4]
```diff
@@ -37,3 +44,41 @@ TEST_F(PortParseTest, InvalidPortConfig) {
     EXPECT_FALSE(getPortNumber("Port 80800;"));
 }
 
+class HandlerParseTest : public ::testing::Test {
+protected:
+    // Test to check the number of handlers parsed from config file
+    // Returns true iff at least one handler is parsed
+    bool getNumberOfHandlersParsed(std::string config_contents) {
```
Comment (Wizzard033) [inline server_config_parser_test.cc]: Could capitalize Get for consistency

**Comment by alfredlucero** [comment]
I'll take note of the changes about port validation and style changes for the next iteration 👍

**Pull request merged**